core/src/main/scala/better/files/UnicodeCharset.scala

```diff
@@ -51,7 +52,8 @@ class UnicodeDecoder(defaultCharset: Charset) extends CharsetDecoder(defaultChar
     }
   }
 
-  override def decodeLoop(in: ByteBuffer, out: CharBuffer) = decode(in = in, out = out, candidates = bomTable.keySet)
+  override def decodeLoop(in: ByteBuffer | Null, out: CharBuffer | Null) =
```

Comment:
**Overridden java method arguments are nullable**
A Scala method overrides some Java method. We need to add `|Null` to the method's parameter types to make the override still work.